docs/development/codeguide.rst

```diff
@@ -134,10 +136,35 @@ Coding Style/Conventions
   using only 4 spaces for indentation, and never tabs.
 
 * Our testing infrastructure currently enforces a subset of the PEP8 style
-  guide. You can check locally whether your changes have followed these by
-  running the following `tox <https://tox.readthedocs.io/>`__ command::
+  guide, and some packages enforce stronger styling checks such as using
+  `isort <https://pycqa.github.io/isort/>`_ to sort the module imports.
+
+  * We provide a `pre-commit <https://pre-commit.com/>`_ based git ``pre-commit``
+    hook, which enforces all of these styling checks on each commit and automatically
+    fixes styling issues whenever possible. Note that ``pre-commit``
+    `hooks <https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks#_git_hooks>`_
+    in git are essentially scripts that get run automatically by git before the
+    commit message can be entered; meaning, if the ``pre-commit`` hook fails you
+    will be unable to commit your code without overriding (which should be avoided
+    whenever possible) the hook, which can be done by running::
+
+      git commit --no-verify
+
+    We strongly suggest that you setup and use the pre-commit hook, by running::
```

Comment:
In general avoid putting code blocks in the middle of a sentence, this is difficult to read. Instead:
```
We encourage you to setup and use the pre-commit hook by running the following 
in your clone of astropy::

  ...

See  the ``pre-commit`` `install guide <https://pre-commit.com/#install>`_ for more details. 
```